interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -1760,33 +1768,38 @@ public void performSoakTest(
             .newBlockingStub(soakChannel)
             .withInterceptors(recordClientCallInterceptor(clientCallCapture));
       }
-      SoakIterationResult result = 
-          performOneSoakIteration(soakStub, soakRequestSize, soakResponseSize);
-      SocketAddress peer = clientCallCapture
-          .get().getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR);
-      StringBuilder logStr = new StringBuilder(
-          String.format(
-              Locale.US,
-              "soak iteration: %d elapsed_ms: %d peer: %s server_uri: %s",
-              i, result.getLatencyMs(), peer != null ? peer.toString() : "null", serverUri));
-      if (!result.getStatus().equals(Status.OK)) {
-        totalFailures++;
-        logStr.append(String.format(" failed: %s", result.getStatus()));
-      } else if (result.getLatencyMs() > maxAcceptablePerIterationLatencyMs) {
-        totalFailures++;
-        logStr.append(
-            " exceeds max acceptable latency: " + maxAcceptablePerIterationLatencyMs);
+
+      final TestServiceGrpc.TestServiceBlockingStub currSoakStub = soakStub;
+      if (concurrent) {
+        semaphore.acquire();
+        // Create a new thread for each soak iteration
```

Comment:
Rather than create a new thread per iteration, I think it would be better to create `--num_threads` threads at the beginning of the soak test, and then have each thread independently run the soak test loop (e.g. run `--soak_iterations` RPCs).